client/hotrod-client/src/test/java/org/infinispan/client/hotrod/query/RemoteQueryDslConditionsTunedTest.java

```diff
@@ -29,23 +30,26 @@ protected ConfigurationBuilder getConfigurationBuilder() {
              .addIndexedEntity("sample_bank_account.User")
              .addIndexedEntity("sample_bank_account.Account")
              .addIndexedEntity("sample_bank_account.Transaction")
-             .addProperty("default.indexmanager", "near-real-time")
-             .addProperty("default.indexBase", indexDirectory)
-             .addProperty("default.exclusive_index_use", "true")
-             .addProperty("default.indexwriter.merge_factor", "30")
-             .addProperty("default.indexwriter.merge_max_size", "4096")
-             .addProperty("default.indexwriter.ram_buffer_size", "220")
-             .addProperty("default.locking_strategy", "native")
-             .addProperty("default.sharding_strategy.nbr_of_shards", String.valueOf(NUM_SHARDS));
+             .addProperty(SearchConfig.DIRECTORY_TYPE, SearchConfig.FILE)
+             .addProperty(SearchConfig.DIRECTORY_ROOT, indexDirectory)
+             .addProperty(SearchConfig.THREAD_POOL_SIZE, String.valueOf(NUM_SHARDS))
+             .addProperty(SearchConfig.QUEUE_COUNT, String.valueOf(NUM_SHARDS))
+             .addProperty(SearchConfig.QUEUE_SIZE, "4096")
+             .addProperty(SearchConfig.COMMIT_INTERVAL, "10000")
+             .addProperty(SearchConfig.SHARDING_STRATEGY, SearchConfig.HASH)
+             .addProperty(SearchConfig.NUMBER_OF_SHARDS, String.valueOf(NUM_SHARDS));
```

Comment:
I think we discussed this already a few months ago, but let's summarize it here.

NRT is enabled by default, sort of. Commits are asynchronous by default and index readers are aware of uncommitted changes by default.

Configuration in Hibernate Search 6 is different from Hibernate Search 5:

* When you request indexing, you (Infinispan) can request that the commit happens immediately.
  * If you do, the commit will happen before the indexing operation returns. Obviously, it will perform badly if you index your documents one by one.
  * If you don't the commit will happen some time after the indexing operation returns. By default it's 1000ms, but you can set it to a different number.

Readers are always aware of uncommitted changes by default, i.e. they are aware of changes still in the index writer buffers.

If you can accept out-of-date readers, you can set a refresh interval so that readers are only re-created every 1000ms, for example.

More information: https://docs.jboss.org/hibernate/search/6.0/reference/en-US/html_single/#backend-lucene-io

Reply:
> is that still the case in Hibernate Search 6?

If you force the commit, it will happen ASAP, so no.

If you don't force the commit, then Hibernate Search will still return before the commit. However, if you set commit interval is set to 0, it will commit just after each "batch" of operations. The size of batches can be configured.

If you want guarantees that changes are flushed to disk, that's obviously not enough. Depending on Infinispan's needs, we could introduce another behavior where you don't force the commit, but ask Hibernate Search to only consider the indexing operation completed when the next commit happens (whenever that is). I believe that shouldn't be too much work. That will have to wait until after this PR is merged, though (we need another release of Search).

Reply:
Hold on a sec. So basically, we are moving from

1) ```master```, where NRT is not enabled by default, and will commit on every document synchronously, but will coalesce writing threads to reduce the number of commits. Once the cache operation completes, it is guarantee the document is indexed on disk and it's visible on searches.

to

2) ```ISPN-10249``` (this PR), where NRT is ON by default, commits are done by batch size or time, cache operations guarantee immediate visibility on searches but not immediate persistence of the index, but only once every second (configurable)

Correct?

If so, I reckon 2) is overall better than 1), so I wouldn't rush implementing the old behaviour

Reply:
That's

Reply:
Correct.

The only drawback of the new behavior is there is no way to get a guarantee that each change is on disk after the cache operation completes unless you accept bad performance (one commit per cache operation).

I believe that kind of requirement is a bit unreasonable for a secondary source of truth such as Lucene indexes, though. We will probably have a look at some time in the future, for the sake of completeness, but I agree with you this can wait.